README.md

```diff
@@ -29,95 +33,319 @@ are well established for RSE-ops, and which require further development. We can
+> everything has tests).
+
+> Internal codes are tested a lot -- there are devlops teams and dedicated clusters for them. Codes
+> also go through V&V process (verification and validation).
 
 ### What about dependency management?
 
-Dependency management refers to a set of practices used to manage software versions. It is arguably a subtopic of software distribution. Every piece of scientific software requires a specific set of versioned dependencies, and these dependencies must co-exist alongside one another on a shared resource. This also means that architectures must be maintained for possibly older dependencies that require them. Tools have grown out of this need that allow for flexible management of dependencies and software, including easybuild [18], spack [19], and environment modules [20,21]. Development of these projects has also led to a powerful model of development -- bringing many people (administrators and users) together to collaborate on the software. It's not hard to read user surveys [22] and see that the open source model of development is live and well. People are using the software, excited about it, and working together to make it better. It also doesn't hurt that some of these projects have the backing of entire institutions. These success stories give us a hint that working together on software, likely in an open source, collaboration fashion, is a good model for successful distribution and improvement of the software.
+Dependency management refers to a set of practices used to manage software versions. It is arguably a subtopic of software distribution. Every piece of scientific software requires a specific set of versioned dependencies, and these dependencies must co-exist alongside one another on a shared resource. This also means that architectures must be maintained for possibly older dependencies that require them. Tools have grown out of this need that allow for flexible management of dependencies and software, including easybuild [18], spack [19], and environment modules [20,21]. Development of these projects has also led to a powerful model of development -- bringing many people (administrators and users) together to collaborate on the software. It's not hard to read user surveys [22] and see that the open source model of development is live and well. People are using the software, excited about it, and working together to make it better. It also doesn't hurt that some of these projects have the backing of entire institutions.
+
+> and programs like ASC and ECP (really it's these -- large LLNL funding is more recent with BUILD
+
+These success stories give us a hint that working together on software, likely in an open source, collaboration fashion, is a good model for successful distribution and improvement of the software.
+
+> the counterargument you'll get (which I kind of disagree with) is that package managers are
```

Comment:
I don't think the necessity of a project (or someone thinking one is more important than another) is really a factor here. If there is some niche community of people excited about something they are going to work on it, and if they do so openly, collaboratively, that's a good/successful model to make the software better.

Reply:
Sure but for smaller communities people make the argument that there just aren't enough people to get critical mass.  The point is that it's worth saying something like this -- that collaboration is beneficial even for small projects.